src/main/resources/testchipip/csrc/mm.cc

```diff
@@ -38,19 +43,36 @@ void mm_t::init(size_t sz, int wsz, int lsz)
   assert(wsz > 0 && lsz > 0 && (lsz & (lsz-1)) == 0 && lsz % wsz == 0);
   word_size = wsz;
   line_size = lsz;
+  seed = std::chrono::system_clock::now().time_since_epoch().count();
+  std::cout << "didn't supply a seed! Random seed is " << seed << std::endl;
+  mt = std::mt19937(seed);
+  std::string path_name = "chipyard-cosim-" + std::to_string(getpid());
```

Comment:
The fd for the shared memory object should not contain references to cosim. Name this something like `mm_<pid>`